src/docs/framework-integration/vue.md

```diff
@@ -8,50 +8,378 @@ contributors:
   - kensodemann
   - brysalazar12
   - iskanderbroere
+  - sean-perkins
 ---
 
 # Vue
 
-In order to use the custom element library within the Vue app, the application must be modified to define the custom elements and to inform the Vue compiler which elements to ignore during compilation. This can all be done within the `main.js` file.
+**Supports: Vue 3 • TypeScript 4.0+ • Stencil v2.9.0+**
 
-Assuming you’ve run `npm install --save test-components` beforehand, and that `test-component` is the name of our made up Web Components that we have published to npm, you import the components into the 'main.js' file by
+Stencil can generate Vue component wrappers for your web components. This allows your Stencil components to be used within a Vue 3 application. The benefits of using Stencil's component wrappers over the standard web components include:
 
-- importing the node module
-- telling Vue to ignore the custom element tags (see [docs](https://vuejs.org/v2/api/#ignoredElements))
-- binding the Stenciljs component code to the window object
+- Type checking with your components.
+- Integration with the router link and Vue router.
+- Optionally, form control web components can be used with `v-model`.
 
-```tsx
-import Vue from 'vue';
-import App from './App.vue';
+## Setup
 
-import { applyPolyfills, defineCustomElements } from 'test-components/loader';
+### Project Structure
 
-Vue.config.productionTip = false;
+We recommend using a monorepo structure for your component library with component wrappers. Your project workspace should contain your Stencil component library and the library for the generate Vue component wrappers.
 
-// Tell Vue to ignore all components defined in the test-components
-// package. The regex assumes all components names are prefixed
-// 'test'
-Vue.config.ignoredElements = [/test-\w*/];
+An example project set-up may look similar to:
 
-// Bind the custom elements to the window object
-applyPolyfills().then(() => {
-  defineCustomElements();
-});
+```
+top-most-directory/
+└── packages/
+    ├── vue-library/
+    │   └── src/
+    │       ├── lib/
+    │       └── index.ts
+    └── stencil-library/
+        ├── stencil.config.js
+        └── src/components
+```
+
+#### Creating a Stencil Component Library
+
+> If you already have a Stencil component library, skip this section.
+
+```bash
+cd packages/
+npm init stencil components stencil-library
+cd stencil-library
+# Install dependencies
+npm install
+# of if you are using yarn
+yarn install
+cd ..
+lerna bootstrap
+```
+
+#### Creating a Vue Component Library
+
+> If you already have a Vue component library, skip this section.
+
+The first time you want to create the component wrappers, you will need to have a Vue library package to write to.
+
+Using Lerna and Vue's CLI, generate a workspace and a library for your Vue component wrappers:
+
+```bash
+lerna init
+npm install typescript @types/node --save-dev
```

Comment:
We may want to clarify that these should be run from the `top-most-directory/` - as a user I had to infer that from [this line](https://github.com/ionic-team/stencil-site/pull/864/files#diff-af403489927c5c8dffa93fe5aa0478c28408f6b160f7e3e029475f676a005321R76), which felt like it could have been a little smoother